app/src/common/shared/com/igalia/wolvic/ui/views/TabsBarItem.java

```diff
@@ -160,6 +158,17 @@ public void onLocationChange(@NonNull WSession session, @Nullable String url) {
         }
     }
 
+    private String getTitleForDisplay(String url, String title) {
+        Windows.ContentType contentType = UrlUtils.getContentType(url);
+        if (contentType != Windows.ContentType.WEB_CONTENT) {
+            return getContext().getString(contentType.titleResId);
+        } else if (StringUtils.isEmpty(title)) {
```

Comment:
No need for else if we are returning when the if condition is satisfied.

Reply:
Updated.